BaristaSwift/Common/Common/Store/LocalCartStore.swift

```diff
@@ -205,131 +184,105 @@ public class LocalCartStore {
-                if let complete = syncState?.isDone(), complete == true {
-                    SalesforceSwiftLogger.log(type(of:self), level:.info, message:"create new opportunity - sync completed")
-                    guard let synced = OpportunityStore.instance.record(forExternalId: optyId) else {
-                        completion(nil)
-                        return
-                    }
-                    completion(synced)
-                }
-            })
+            SalesforceSwiftLogger.log(type(of:self), level:.info, message:"create new opportunity")
+            return OpportunityStore.instance.createEntry(entry: newOpty)
         } else {
             SalesforceSwiftLogger.log(type(of:self), level:.info, message:"returning existing opportunity")
-            completion(inProgress.first!)
+            return Promise.value(opptyInProgress.first!)
         }
     }
     
-    fileprivate func getOrCreateNewQuote(forOpportunity opportunity:Opportunity, withAccount account:Account, completion:@escaping (Quote?) -> Void) {
+    fileprivate func getOrCreateNewQuote(forOpportunity opportunity:Opportunity, withAccount account:Account) -> Promise<Quote> {
         SalesforceSwiftLogger.log(type(of:self), level:.info, message:"getOrCreateNewQuote")
         // assign opportunity primary quote and sync up
         if let primary = opportunity.primaryQuote, let quote = QuoteStore.instance.quoteFromId(primary) {
             SalesforceSwiftLogger.log(type(of:self), level:.info, message:"returning exisitng quote")
-            completion(quote)
+            return Promise.value(quote)
         } else {
-            let newQuote = Quote()
+            var newQuote = Quote()
             newQuote.primaryQuote = true
             newQuote.opportunity = opportunity.id
             newQuote.account = account.accountId
             newQuote.pricebookId = opportunity.pricebookId
             newQuote.lineItemsGrouped = true
             newQuote.primaryQuote = true
-            let newQuoteId = newQuote.externalId
             SalesforceSwiftLogger.log(type(of:self), level:.info, message:"creating new quote")
-            QuoteStore.instance.create(newQuote, completion: { (syncState) in
-                // Todo - need to handle sync failure properly
-                if let complete = syncState?.isDone(), complete == true {
-                    SalesforceSwiftLogger.log(type(of:self), level:.info, message:"create new quote - sync completed")
-                    guard let synced = QuoteStore.instance.record(forExternalId: newQuoteId) else {
-                        completion(nil)
-                        return
-                    }
-                    opportunity.primaryQuote = synced.quoteId
-                    OpportunityStore.instance.upsertEntries(record: opportunity)
-                    OpportunityStore.instance.syncUp()
-                    completion(synced)
```

Comment:
The old code would fire the completing code without waiting for the opportunity sync up.
I changed that for now, but it might look slower as a result.